dtu_mlops_project/models/train_model.py

```diff
@@ -1,13 +1,61 @@
-import torch
+import os
+import logging
+import dotenv
+import hydra
+import lightning as L
 
 
-def train_model():
-    print("Training model...")
-    print(f"PyTorch Version: {torch.__version__}")
-    ...
-    print("Model trained!")
+@hydra.main(version_base="1.3", config_path="../configs", config_name="train.yaml")
+def main(cfg):
+    logger = logging.getLogger(__name__)
+    cwd = os.path.abspath(os.getcwd())
+    logger.info(f"Current experiment directory: {cwd}")
 
+    if cfg.get("seed"):
+        logger.info(f"Setting seed to {cfg.seed}")
+        L.seed_everything(cfg.seed, workers=True)
+
+    logger.info(f"Instantiating datamodule <{cfg.datamodule._target_}>")
+    datamodule = hydra.utils.instantiate(cfg.datamodule)
+
+    logger.info(f"Instantiating model <{cfg.model._target_}>")
+    model = hydra.utils.instantiate(cfg.model)
+
+    # logger.info("Instantiating callbacks...")
+    # callbacks = instantiate_callbacks(cfg.get("callbacks"))
+    callbacks = []
+
+    logger.info(f"Instantiating logger <{cfg.logger._target_}>")
+    pl_logger = hydra.utils.instantiate(cfg.logger)
+
+
+    logger.info(f"Instantiating trainer <{cfg.trainer._target_}>")
+    trainer = hydra.utils.instantiate(cfg.trainer, callbacks=callbacks, logger=pl_logger)
```

Comment:
I maybe don't get it but does this automatically create an instance of pytorch Trainer, based simply on the config file?

Reply:
yes